include/mbedtls/ecdh.h

```diff
@@ -63,6 +93,15 @@ typedef struct mbedtls_ecdh_context
     mbedtls_ecp_point Vi;    /*!< The blinding value. */
     mbedtls_ecp_point Vf;    /*!< The unblinding value. */
     mbedtls_mpi _d;          /*!< The previous \p d. */
+#else
+    int point_format;        /*!< The format of point export in TLS messages. */
```

Comment:
In new data structures, please refrain from using `int`. Use a suitable type instead: `size_t` for sizes, an unsigned type for most integers, an `enum` when only a small number of cases are possible... If I'm not mistaken, the point format is a 16-bit value, so the correct type would be `uint_least16_t` or, until we fully update our build chains to C99, `unsigned short`.

Reply:
We are already using `uint16_t` in both the ECP and the SSL modules, therefore I think it would be safe to use (at least it wouldn't make things worse). Theoretically `uint_least16_t` would be nice, but it is not the most frequently used feature, are you sure it is worth the extra trust we need to put in our toolchains?

Reply:
(It turned out that it is an 8-bit value, but we have `uint8_t` too in the SSL module already so the same argument holds. Also it is much less likely to have a platform without `uint8_t` than one without `uint16_t`)

Reply:
I think `stdint.h` is already allowed (it's the only C99 feature that's allowed so far). It's currently included in 26 places in the library, so I really don't see a reason to refrain from including it here.

Also, since we require `CHAR_BIT == 8`, I don't really see a reason to bother with the `uint_leastXX_t` types rather than using `uintXX_t` directly, as in practice all platforms that have the former and 8-bit bytes are going to have the later too. (Also, `grep 'int[0-9][0-9]*_t' library/*.c include/mbedtls/*.h | wc -l` prints `580` so if those types were a problem of our users, we would already know.)